File: src/atcoder/abc314/ruby/abc314_c.rb
n, m = gets.split.map(&:to_i)
s = gets.chomp
c = gets.split.map(&:to_i).map { _1.pred }

ix = Array.new(m) { [] }
n.times do |i|
  ix[c[i]] << i
end

ans = Array.new(n, -1)
m.times do |i|
  row = ix[i]
  k = row.size
  k.times do |j|
    jj = (j + 1) % k
    ans[row[jj]] = row[j]
  end
end

puts ans.map { s[_1] }.join

File: src/atcoder/abc314/python3/abc314_c.py
n, m = [int(s) for s in input().split()]
s = [c for c in input()]
c = [int(s) - 1 for s in input().split()]

ix = [[] for _ in range(m)]
for i in range(n):
  ix[c[i]].append(i)

ans = [-1 for _ in range(n)]
for i in range(m):
  row = ix[i]
  k = len(row)
  for j in range(k):
    jj = (j + 1) % k
    ans[row[jj]] = row[j]

print("".join([s[i] for i in ans]))